### PR TITLE
replay: upsert to db

### PR DESCRIPTION
**Pull request by h4x3rotab**
This PR allows the replayer to override dirty data. Sometimes if we run an incorrect version of replayer, it may write bad data to the db. In such case, we can manually remove the bad data an rerun GK replay. However, without this PR, the write the db will be batched, and some rows at the edge may conflict with the data left in the db, and causing the full batch to fail to write. With this PR, the write will always override the rows in the database.

### CODE REVIEW

File: standalone/replay/src/replay_gk/data_persist.rs [hunk 1]
```diff
@@ -116,7 +116,11 @@ async fn insert_records(pool: &sqlx::Pool<sqlx::Postgres>, records: &[EventRecor
         INSERT INTO worker_finance_events
             (sequence, pubkey, block, time, event, v, p, payout)
         SELECT *
-        FROM UNNEST($1, $2, $3, $4, $5, $6, $7, $8)
+        FROM UNNEST($1, $2, $3, $4, $5, $6, $7, $8) u
```
[the remark this answers is not in the document]
Comment (h4x3rotab) [reply standalone/replay/src/replay_gk/data_persist.rs:119]: Will remove. It's an alias not in used

File: standalone/replay/src/replay_gk/data_persist.rs [hunk 2]
```diff
@@ -116,7 +116,11 @@ async fn insert_records(pool: &sqlx::Pool<sqlx::Postgres>, records: &[EventRecor
...
+        ON CONFLICT (time, sequence)
```
[the remark this answers is not in the document]
Comment (h4x3rotab) [reply standalone/replay/src/replay_gk/data_persist.rs:120]: Updated the description

File: standalone/replay/src/replay_gk/data_persist.rs [hunk 3]
```diff
@@ -116,7 +116,11 @@ async fn insert_records(pool: &sqlx::Pool<sqlx::Postgres>, records: &[EventRecor
...
+        DO UPDATE
+        SET (pubkey, block, event, v, p, payout)
+            = (excluded.pubkey, excluded.block, excluded.event, excluded.v, excluded.p, excluded.payout)
```
[the remark this answers is not in the document]
Comment (h4x3rotab) [reply standalone/replay/src/replay_gk/data_persist.rs]: Sounds good. Fixed

**Review by kvinwang — APPROVED** [review]
LGTM

**Pull request merged**